<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Fits linear ridge regression models for genome-wide SNP data. — linearRidgeGenotypes • ridge</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Fits linear ridge regression models for genome-wide SNP data. — linearRidgeGenotypes"><meta property="og:description" content="Fits linear ridge regression models for genome-wide SNP data. The SNP
  genotypes are not read into R but file names are passed the code
  directly, enabling the analysis of genome-wide scale SNP data sets."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ridge</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/SteffenMoritz/ridge/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Fits linear ridge regression models for genome-wide SNP data.</h1>
    
    <div class="hidden name"><code>linearRidgeGenotypes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Fits linear ridge regression models for genome-wide SNP data. The SNP
  genotypes are not read into R but file names are passed the code
  directly, enabling the analysis of genome-wide scale SNP data sets.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">linearRidgeGenotypes</span><span class="op">(</span><span class="va">genotypesfilename</span>, <span class="va">phenotypesfilename</span>, lambda <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, 
thinfilename <span class="op">=</span> <span class="cn">NULL</span>, betafilename <span class="op">=</span> <span class="cn">NULL</span>, approxfilename <span class="op">=</span> <span class="cn">NULL</span>,
permfilename <span class="op">=</span> <span class="cn">NULL</span>, intercept <span class="op">=</span>
<span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>genotypesfilename</dt>
<dd><p>character string: path to file containing SNP genotypes coded 0, 1,
  2. See <code>Input file formats</code>.</p></dd>
<dt>phenotypesfilename</dt>
<dd><p>character string: path to file containing phenotypes. See <code>Input file formats</code>.</p></dd>
<dt>lambda</dt>
<dd><p>(optional) shrinkage parameter. If not provided, the default denotes
  automatic choice of the shrinkage parameter using the method of Cule &amp;
  De Iorio (2012).</p></dd>
<dt>thinfilename</dt>
<dd><p>(optional) character string: path to file containing three columns: SNP name, chromosme and SNP psotion. See <code>Input file formats</code>. (See <code>details</code>.)</p></dd>
<dt>betafilename</dt>
<dd><p>(optional) character string: path to file where the output will be written. See <code>Output file formats</code>.</p></dd>
<dt>approxfilename</dt>
<dd><p>(optional) character string: path to fine where the approximate test p-values will be written. 
Approximate p-values are not computed unless this argument is given. Approximate p-values
  are computed using the method of Cule et al (2011). See <code>Output file formats</code>.</p></dd>
<dt>permfilename</dt>
<dd><p>(optional) character string: path to file where the permutation test
  p-values will be written.
  Permutation test p-values are not computed unless this argument is
  given. (See warning). See <code>Output file formats</code>.</p></dd>
<dt>intercept</dt>
<dd><p>Logical: Should the ridge regression model be fitted with an
  intercept? (Defaults to <code>TRUE</code>)</p></dd>
<dt>verbose</dt>
<dd><p>Logical: If <code>TRUE</code>, additional information is printed to the R
  output as the code runs. Defaults to <code>FALSE</code>.</p></dd>
</dl></div>
    <div id="input-file-formats">
    <h2>Input file formats</h2>
    
<dl><dt>genotypesfilename:</dt>
<dd><p>A header row, plus one row for each
  individual, one SNP per column. The header row contains SNP
  names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
  values are not accommodated. Invariant SNPs in the data cause an
  error, please remove these from the file before calling the function.</p></dd>

<dt>phenofilename:</dt>
<dd><p>A single column of phenotypes with the individuals in the same order as those in the file <code>genotypesfilename</code>.</p></dd>

<dt>thin:</dt>
<dd><p>(optional) Three columns and the same number of rows as there are SNPs in the file <code>genotypesfilename</code>, one row per SNP. First column: SNP names (must match names in <code>genotypesfilename</code>); second column: chromosome;  third column: SNP position in BP.</p></dd>


</dl></div>
    <div id="output-file-formats">
    <h2>Output file formats</h2>
    <p>All output files are optional. Whether or not <code>betafilename</code> is provided, fitted coefficients are returned to the R workshpace. If <code>betafilename</code> is provided, fitted coefficients are written to the file specified (in addition).</p><dl><dt>betafilename:</dt>
<dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If <code>intercept = TRUE</code> (the default) then the first row is the fitted intercept (with the name Intercept in the first column).</p></dd>

<dt>approxfilename:</dt>
<dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is approximate p-values.</p></dd>

<dt>permfilename:</dt>
<dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is permutation p-values.</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>If a file <code>thin</code> is supplied, and the shrinkage parameter
  <code>lambda</code> is being computed automatically based on the data, then
  this file is used to thin the SNP data by SNP position. If this file
  is not supplied, SNPs are thinned automatically based on number of SNPs.</p>
    </div>
    <div id="value">
    <h2>Value</h2>
    <p>The vector of fitted ridge regression coefficients.
  If <code>betafilename</code> is given, the fitted coefficients are written to this
  file as well as being returned.
  If <code>approxfilename</code> and/or <code>permfilename</code> are given, results of approximate
  test p-values and/or permutation test p-values are written to the files
  given in their arguments.</p>
    </div>
    <div id="warning-">
    <h2>Warning </h2>
    <p>When data are large, the permutation test p-values
  may take a very long time to compute. It is recommended not to request
  permutation test p-values (using the argument <code>permfilename</code>)
  when data are large.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Significance testing in ridge regression for genetic data. Cule, E. et
  al (2011) BMC Bioinformatics, 12:372
  A semi-automatic method to guide the choice of ridge parameter in
  ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP]</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Erika Cule</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="linearRidge.html">linearRidge</a></code> for fitting linear ridge regression models
  when the data are small enough to be read into R.
  <code><a href="logisticRidge.html">logisticRidge</a></code> and <code><a href="logisticRidgeGenotypes.html">logisticRidgeGenotypes</a></code> for fitting logistic ridge
  regression models.</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in">    <span class="va">genotypesfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"GenCont_genotypes.txt"</span>,package <span class="op">=</span> <span class="st">"ridge"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">phenotypesfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"GenCont_phenotypes.txt"</span>,package <span class="op">=</span> <span class="st">"ridge"</span><span class="op">)</span></span>
<span class="r-in">    <span class="va">beta_linearRidgeGenotypes</span> <span class="op">&lt;-</span> <span class="fu">linearRidgeGenotypes</span><span class="op">(</span>genotypesfilename <span class="op">=</span> <span class="va">genotypesfile</span>,</span>
<span class="r-in">phenotypesfilename <span class="op">=</span> <span class="va">phenotypesfile</span><span class="op">)</span></span>
<span class="r-in">    <span class="co">## compare to output of linearRidge</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GenCont</span><span class="op">)</span> <span class="co">## Same data as in GenCont_genotypes.txt and GenCont_phenotypes.txt</span></span>
<span class="r-in">    <span class="va">beta_linearRidge</span> <span class="op">&lt;-</span> <span class="fu"><a href="linearRidge.html">linearRidge</a></span><span class="op">(</span><span class="va">Phenotypes</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">GenCont</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">beta_linearRidge</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span>, <span class="va">beta_linearRidgeGenotypes</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
<span class="r-in">  </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Steffen Moritz, Erika Cule, Dan Frankowski.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

