<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Predict phenotypes from genome-wide SNP data based on a file of coefficients — linearRidgeGenotypesPredict • ridge</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Predict phenotypes from genome-wide SNP data based on a file of coefficients — linearRidgeGenotypesPredict"><meta property="og:description" content="Predict phenotypes from genome-wide SNP data based on a file of
  coefficients. Genotypes and fitted coefficients are provided as
  filenames, allowing the computation of fitted probabilities when SNP
  data are too large to be read into R."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ridge</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/SteffenMoritz/ridge/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Predict phenotypes from genome-wide SNP data based on a file of coefficients</h1>
    
    <div class="hidden name"><code>linearRidgeGenotypesPredict.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Predict phenotypes from genome-wide SNP data based on a file of
  coefficients. Genotypes and fitted coefficients are provided as
  filenames, allowing the computation of fitted probabilities when SNP
  data are too large to be read into R.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">linearRidgeGenotypesPredict</span><span class="op">(</span><span class="va">genotypesfilename</span>, <span class="va">betafilename</span>, phenotypesfilename <span class="op">=</span> <span class="cn">NULL</span>,
verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>genotypesfilename</dt>
<dd><p>character string: path to file containing SNP genotypes coded 0, 1,
  2. See <code>Input file formats</code>.</p></dd>
<dt>betafilename</dt>
<dd><p>character string: path to file containing fitted coefficients. See <code>Input file formats</code>.</p></dd>
<dt>phenotypesfilename</dt>
<dd><p>(optional) character string: path to file in which to write out the
  predicted phenotypes. See <code>Output file formats</code>. Whether or not this argument
  is supplied, the fitted coefficients are also returned by the function.</p></dd>
<dt>verbose</dt>
<dd><p>Logical: If <code>TRUE</code>, additional information is printed to the R
  outupt as the code runs. Defaults to <code>FALSE</code>.</p></dd>
</dl></div>
    <div id="input-file-formats">
    <h2>Input file formats</h2>
    
<dl><dt>genotypesfilename:</dt>
<dd><p>A header row, plus one row for each
  individual, one SNP per column. The header row contains SNP
  names. SNPs are coded as 0, 1, 2 for minor allele count. Missing
  values are not accommodated.</p></dd>

<dt>betafilename:</dt>
<dd><p>Two columns: First column is SNP names in same order as in <code>genotypesfilename</code>, second column is fitted coefficients. If the coefficients include an intercept then the first row of <code>betafilename</code> should contain it with the name Intercept in the first column. An Intercept thus labelled will be used appropriately in predicting the phenotypes. SNP names must match those in <code>genotypesfilename</code>. 
  The format of <code>betafilename</code> is
  that of the output of <code><a href="linearRidgeGenotypes.html">linearRidgeGenotypes</a></code>, meaning
  <code>linearRidgeGenotypesPredict</code> can be used to predict using
  coefficients fitted using <code><a href="linearRidgeGenotypes.html">linearRidgeGenotypes</a></code> (see the example).</p></dd>


</dl></div>
    <div id="output-file-format">
    <h2>Output file format</h2>
    <p>Whether or not <code>phenotypesfilename</code> is provided, predicted phenotypes are returned to the R workshpace. If <code>phenotypesfilename</code> is provided, predicted phenotypes are written to the file specified (in addition).</p><dl><dt>phenotypesfilename:</dt>
<dd><p>One column, containing predicted phenotypes, one individual per row.</p></dd>


</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A vector of fitted values, the same length as the number of
  individuals whose data are in <code>genotypesfilename</code>. If
<code>phenotypesfilename</code> is supplied, the fitted values are also
  written there.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>A semi-automatic method to guide the choice of ridge parameter in ridge regression. Cule, E. and De Iorio, M. (2012) arXiv:1205.0686v1 [stat.AP]</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Erika Cule</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="linearRidgeGenotypes.html">linearRidgeGenotypes</a></code> for model
  fitting. <code><a href="logisticRidgeGenotypes.html">logisticRidgeGenotypes</a></code> and
  <code><a href="logisticRidgeGenotypesPredict.html">logisticRidgeGenotypesPredict</a></code> for corresponding functions
  to fit and predict on SNP data with binary outcomes.</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="va">genotypesfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"GenCont_genotypes.txt"</span>,package <span class="op">=</span> <span class="st">"ridge"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">phenotypesfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"GenCont_phenotypes.txt"</span>,package <span class="op">=</span> <span class="st">"ridge"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">betafile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"beta"</span>, fileext <span class="op">=</span> <span class="st">".dat"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">beta_linearRidgeGenotypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="linearRidgeGenotypes.html">linearRidgeGenotypes</a></span><span class="op">(</span>genotypesfilename <span class="op">=</span> <span class="va">genotypesfile</span>,</span>
<span class="r-in">                                                      phenotypesfilename <span class="op">=</span> <span class="va">phenotypesfile</span>,</span>
<span class="r-in">                                                      betafilename <span class="op">=</span> <span class="va">betafile</span><span class="op">)</span></span>
<span class="r-in"><span class="va">pred_phen_geno</span> <span class="op">&lt;-</span> <span class="fu">linearRidgeGenotypesPredict</span><span class="op">(</span>genotypesfilename <span class="op">=</span> <span class="va">genotypesfile</span>,</span>
<span class="r-in">                                                    betafilename <span class="op">=</span> <span class="va">betafile</span><span class="op">)</span></span>
<span class="r-in"><span class="co">## compare to output of linearRidge</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GenCont</span><span class="op">)</span> <span class="co">## Same data as in GenCont_genotypes.txt and GenCont_phenotypes.txt</span></span>
<span class="r-in"><span class="va">beta_linearRidge</span> <span class="op">&lt;-</span> <span class="fu"><a href="linearRidge.html">linearRidge</a></span><span class="op">(</span><span class="va">Phenotypes</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">GenCont</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">pred_phen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">beta_linearRidge</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">pred_phen_geno</span>, <span class="va">pred_phen</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="co">## Delete the temporary betafile</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">betafile</span><span class="op">)</span></span>
<span class="r-in"><span class="op">}</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Steffen Moritz, Erika Cule, Dan Frankowski.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

